mlflow/utils/annotations.py

```diff
@@ -0,0 +1,28 @@
+def experimental(func):
+    """
+    Decorator for marking APIs experimental in the docstring.
+
+    :param func: A function to mark
+    :returns Decorated function.
+    """
+    notice = ".. Note:: Experimental: This method may change or " + \
+             "be removed in a future release without warning.\n"
+    func.__doc__ = notice + func.__doc__
+    return func
+
+
+def deprecated(alternative=None):
```

Comment:
Should we add a `since`? I think this helps give users a notion of when it might be removed. e.g., `since 1.0` suggests it may be removed in 1.1